complete intellij quick start guide with some extra steps (#5653)
geoserver/geoserver: doc/en/developer/source/quickstart/intellij.rst

   .. image:: img/intellij_import_finish.png
      :width: 600

Before Running Geoserver
---------------------------

> GeoServer relies on other libraries that are manatined in parallel under the same project umbrella. These are Geotools and GeoWebCache. So for easier installation and setup of the GeoServer development environment you need to apply the following:

Reviewer: Not strictly mandatory, although it helps to have freshly built jars. locally.

```suggestion
GeoServer relies on other libraries that are manatined in parallel under the same project umbrella. These are Geotools and GeoWebCache. So for easier installation and setup of the GeoServer development environment you may want to apply the following:
```

> GeoServer relies on other libraries that are manatined in parallel under the same project umbrella. These are Geotools and GeoWebCache. So for easier installation and setup of the GeoServer development environment you may want to apply the following:

Reviewer: ```suggestion
GeoServer relies on other libraries that are maintained in parallel under the same project umbrella. These are Geotools and GeoWebCache. So for easier installation and setup of the GeoServer development environment, you may want to apply the following:
```

> #. Download the code of both and execute mvn clean install to them.

Reviewer: ```suggestion
#. Download the code of both and execute ``mvn clean install`` into them.
```

          :width: 400

    #. If you want to use ``errorprone``, notably to perform the QA checks, install the ``Error Prone Compiler`` plugin, restart the IDE and set ``Javac with error-prone`` as a default compiler for the project. Please note that this will slower the build.

> #. If there are errors such as "cannot find symbol AbstractUserGroupServiceTest", rebuild the security-tests project in the security module.  Right click on the security-tests project and click Rebuild.

Reviewer: ```suggestion
#. If there are errors such as "cannot find symbol AbstractUserGroupServiceTest", rebuild the ``security-tests`` project in the security module.  Right-click on the ``security-tests`` project and click Rebuild.
```

#. In the last versions of Intellij Annotations processors are enabled. If there are errors because of this uncheck this options from compiler settings.

>     .. image::img/intellij_disable_annotation_processors.jpeg

Reviewer 2: This image directive is indented by one character (so it will appear as a block quote)

--------------------------------------

#. Add bash to your Windows environment path and restart Intellij.  

> #. If there are errors such as "cannot find symbol AbstractUserGroupServiceTest", rebuild the security-tests project in the security module.  Right click on the security-tests project and click Rebuild.

Reviewer: These two are not really Windows specific.

Reviewer: Here I mean, they are good comments, but don't belong in the Windows specific section, should be above, in the general introduction.

Author: Moved in the latest commit. So this should appear as outdated
